designer/client/src/PageEdit.tsx

```diff
@@ -84,19 +86,18 @@ export class PageEdit extends Component<Props, State> {
       return
     }
 
-    let copy = { ...data }
-    const [copyPage, copyIndex] = findPage(data, page.path)
+    let copy = structuredClone(data)
+    const pageEdit = findPage(copy, page.path)
 
     if (payload.path !== page.path) {
-      copy = updateLinksTo(data, page.path, payload.path)
-      copyPage.path = payload.path
+      copy = updateLinksTo(copy, pageEdit, {
+        path: payload.path
+      })
     }
 
-    copyPage.title = payload.title
-    copyPage.controller = controller
-    copyPage.section = section?.name
-
-    copy.pages[copyIndex] = copyPage
+    pageEdit.title = payload.title
+    pageEdit.controller = controller
+    pageEdit.section = section?.name
```

Comment:
I'm going to move these lines above `updateLinksTo()`

This is because `pageEdit` becomes stale when `definition` is cloned